src/MerQure.RbMQ/MessagingService.cs

```diff
@@ -7,37 +9,26 @@ namespace MerQure.RbMQ
 {
     public class MessagingService : IMessagingService
     {
-        private static IConnection GetRabbitMqConnection()
-        {
-            var rabbitMqConnection = Config.RabbitMqConfiguration.GetConfig().Connection;
-
-            ConnectionFactory connectionFactory = new ConnectionFactory
-            {
-                Uri = rabbitMqConnection.ConnectionString,
-                AutomaticRecoveryEnabled = rabbitMqConnection.AutomaticRecoveryEnabled,
-                TopologyRecoveryEnabled = rabbitMqConnection.TopologyRecoveryEnabled
-            };
-
-            return connectionFactory.CreateConnection();
-        }
-        private static readonly Lazy<IConnection> currentConnection = new Lazy<IConnection>(GetRabbitMqConnection);
-        public static IConnection CurrentConnection => currentConnection.Value;
-
-        public bool Durable { get; private set; }
-        public bool AutoDeleteQueue { get; private set; }
-        public string ExchangeType { get; private set; }
+        public string ExchangeType { get; set; } = RabbitMQ.Client.ExchangeType.Topic;
+        public bool Durable { get; set; }
+        public bool AutoDeleteQueue { get; set; }
         public ushort DefaultPrefetchCount { get; set; }
         public long PublisherAcknowledgementsTimeoutInMilliseconds { get; set; }
 
-        public MessagingService()
+        protected IConnection CurrentConnection => _sharedConnection.CurrentConnection;
+
+        private readonly SharedConnection _sharedConnection;
+        private readonly IOptions<MerQureConfiguration> _merQureConfiguration;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the '_merQureConfiguration' field and declare it as a local variable in the relevant methods. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS1450)